osu.Game/Skinning/Skin.cs

```diff
@@ -239,5 +246,50 @@ protected virtual void Dispose(bool isDisposing)
         }
 
         #endregion
+
+        public override string ToString() => $"{GetType().ReadableName()}{{ Name: {Name} }}";
+
+        private static readonly ThreadLocal<int> nested_level = new ThreadLocal<int>(() => 0);
+
+        [Conditional("SKIN_LOOKUP_DEBUG")]
+        internal static void LogLookupDebug(object callingClass, object lookup, LookupDebugType type, [CallerMemberName] string callerMethod = "")
```

Comment:
The API design here has me thinking... Every single usage of this passes `this` as the first argument - so maybe we want to make an extension out of this? That way you literally _cannot_ mess that part up.

Reply:
Actually no because that would imply letting the first arg be `object`. Bad bad bad. Disregard.